src/Shared/UnitTests/TestEnvironment.cs

```diff
@@ -407,10 +406,40 @@ public class TransientTestProjectWithFiles : TransientTestState
             CreatedFiles = Helpers.CreateFilesInDirectory(TestRoot, files);
         }
 
+        internal MockLogger BuildProjectExpectFailure(IDictionary<string, string> globalProperties = null, string toolsVersion = null)
```

Comment:
Nitpicking a bit, I guess. Why `internal` and not public? This is a tests only class.

Reply:
Also, are these two wrappers useful? And it's not obvious from the name that this would return a logger.

Reply:
`MockLogger` is internal so the methods had to be internal as well.  These methods reflect other functionality we have in `ObjectModelHelpers`